financial-services-accelerator/internal-webapps/org.wso2.financial.services.accelerator.authentication.endpoint/src/main/java/org/wso2/financial/services/accelerator/authentication/endpoint/FSConsentConfirmServlet.java

```diff
@@ -68,9 +69,22 @@ public class FSConsentConfirmServlet extends HttpServlet {
     // Suppressed warning count - 1
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+        String sessionDataKey = request.getParameter(Constants.SESSION_DATA_KEY_CONSENT);
+        HttpSession session = request.getSession();
+
+        // validating session data key format
+        try {
+            UUID.fromString(sessionDataKey);
+        } catch (IllegalArgumentException e) {
+            log.error("Invalid UUID", e);
+            session.invalidate();
+            response.sendRedirect("retry.do?status=Error&statusMsg=Invalid UUID");
```

Comment:
Invalid Session Data Key right? not UUID.

Reply:
Fixed with d5af130df37b33215f19d51199d5633b3443ede4